src/mudlet.cpp

```diff
@@ -2641,6 +2654,153 @@ void mudlet::slot_replay()
+    return output == "mudlet.desktop";
+#endif
+
+#if defined(Q_OS_MACOS)
+    return true;
+#endif
+    // Unknown operating system
+    // Return true to we don't try to set Mudlet as the default
+    return true;
+}
+
+// open a dialog to prompt the user to set Mudlet as default
+void mudlet::openDefaultCheck()
+{
+    if (!mpDefaultClientDlg) {
+        QUiLoader loader;
+        QFile file(QStringLiteral(":/ui/check_default_client.ui"));
+        file.open(QFile::ReadOnly);
+        mpDefaultClientDlg = dynamic_cast<QDialog*>(loader.load(&file, this));
+        file.close();
+
+        if (!mpDefaultClientDlg) {
+            // don't need to print error message from loader as that is already in stdout
+            qWarning() << "mudlet::openDefaultClientCheck() error: failed to load dialog: " << loader.errorString();
+            return;
+        }
+
+        auto buttonBox = mpDefaultClientDlg->findChild<QDialogButtonBox*>(QStringLiteral("buttonBox"));
+        auto checkBox_alwaysPerformCheck = mpDefaultClientDlg->findChild<QCheckBox*>(QStringLiteral("checkBox_alwaysPerformCheck"));
+
+        buttonBox->clear();
+
+        auto setAsDefault = new QPushButton(tr("Use Mudlet as my default client"), mpDefaultClientDlg);
+        auto notNow = new QPushButton(tr("Not now"), mpDefaultClientDlg);
+
+        buttonBox->addButton(setAsDefault, QDialogButtonBox::AcceptRole);
+        setAsDefault->setAutoDefault(true);
+        buttonBox->addButton(notNow, QDialogButtonBox::RejectRole);
+        connect(buttonBox, &QDialogButtonBox::accepted, mpDefaultClientDlg, &QDialog::accept);
+        connect(buttonBox, &QDialogButtonBox::rejected, mpDefaultClientDlg, &QDialog::reject);
+
+        checkBox_alwaysPerformCheck->setChecked(mudlet::self()->mAlwaysCheckDefault);
+        connect(checkBox_alwaysPerformCheck, &QCheckBox::stateChanged, this, [=](int state) {
+            if (state == Qt::Checked) {
+                mudlet::self()->mAlwaysCheckDefault = true;
+            } else {
+                mudlet::self()->mAlwaysCheckDefault = false;
+            }
+        });
+
+        connect(setAsDefault, &QAbstractButton::clicked, this, [=]() { setMudletAsDefault(); });
+
+        mpDefaultClientDlg->setAttribute(Qt::WA_DeleteOnClose);
+#if defined(Q_OS_WIN32)
+        // framelooks really poor in Windows
+        mpDefaultClientDlg->setWindowFlags(Qt::Tool);
+#else
+        mpDefaultClientDlg->setWindowFlags(Qt::Tool | Qt::FramelessWindowHint);
+#endif
+    }
+
+    mpDefaultClientDlg->raise();
+    mpDefaultClientDlg->show();
+    mpDefaultClientDlg->move(mpDefaultClientDlg->x(), 0);
+}
+
+void mudlet::setMudletAsDefault()
+{
+#if defined(Q_OS_WIN)
+    QSettings settings("HKEY_CLASSES_ROOT", QSettings::NativeFormat);
+    settings.setValue("telnet/.", "URL:Telnet Protocol");
+    settings.setValue("telnet/URL Protocol", "");
+    settings.setValue("telnet/shell/open/command/.", "mudlet %1");
+#endif
+#if defined(Q_OS_LINUX)
```

Comment:
These two blocks of code are mutual exclusive so they should be combined - also as I reported above the latter should work for FreeBSD so it can be combined with the Linux one:
```suggestion
#elif defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
```